utilities/SiteExcelToSql/Program.cs

```diff
@@ -0,0 +1,160 @@
+                {
+                    // skip first 2 header rows
+                    reader.Read();
+                    reader.Read();
+
+                    // start to read data rows on site info sheet
+                    while (reader.Read())
+                    {
+                        siteList.Add(reader.ParseSite());
+                    }
+
+                    // read remote access sheet
+                    reader.NextResult();
+
+                    // skip 2 header rows
+                    reader.Read();
+                    reader.Read();
+
+                    var userCount = 0;
+                    while (reader.Read())
+                    {
+                        var pec = reader.GetString(0);
+                        var user = reader.ParseRemoteUser();
+
+                        if (string.IsNullOrEmpty(pec))
+                        {
+                            continue;
+                        }
+                        userCount++;
+
+                        // assume PEC is unique
+                        var matchingSite = siteList.SingleOrDefault(s => s.PEC == pec);
+                        if (matchingSite == null)
+                        {
+                            Console.WriteLine("No matching site with PEC \"{0}\" for user \"{1} {2}\" at row {3}",
+                                pec, user.FirstName, user.LastName, reader.Depth);
+                        }
+                        else
+                        {
+                            // attach remote user to site
+                            matchingSite.RemoteUsers.Add(user);
+                        }
+                    }
+
+                    PrintStats(siteList, userCount);
+                }
+            }
+
+            var templateSql = File.ReadAllText(Path.Combine(AssemblyDirectory, "template.sql"));
+
+            using var fs = new FileStream(Path.Combine(AssemblyDirectory, outputFileName), FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.None);
+            // clear old file content
+            fs.SetLength(0);
+
+            var writer = new StreamWriter(fs);
+            writer.WriteLine(templateSql);
+
+            // data section
+            for (var i = 0; i < siteList.Count; i++)
+            {
+                // output site SQL
+                writer.WriteLine("\t-- Site {0} - {1} | {2}", i + 1, siteList[i].PEC, siteList[i].DoingBusinessAs);
+                writer.WriteLine("\tINSERT INTO public.\"Site\" (\"CreatedUserId\", \"CreatedTimeStamp\", \"UpdatedUserId\", \"UpdatedTimeStamp\", \"PEC\", \"Completed\", \"DoingBusinessAs\", \"OrganizationId\", \"Status\") "
+                                + "VALUES(vCreatorUUID, now(), vCreatorUUID, now(), '{0}', TRUE, '{1}', vOrganizationID, 1) "
+                                + "RETURNING \"Id\" INTO vSiteId;",
+                                siteList[i].PEC, siteList[i].DoingBusinessAs.Replace("'", "''"));
+
```

Comment:
Shouldn't `CareSettingCode` be populated, even though it is nullable?  It appears to be missing from the spreadsheet?  Can we assume they are all Community Health Practice?  

Also after executing the generated SQL, I see a lot of rows in the database but they don't appear in the /admin interface?  It's not explicitly mentioned in PRIME-1576 but I'm assuming that that is the expectation?

Reply:
> Shouldn't `CareSettingCode` be populated, even though it is nullable? It appears to be missing from the spreadsheet? Can we assume they are all Community Health Practice?
> 
> Also after executing the generated SQL, I see a lot of rows in the database but they don't appear in the /admin interface? It's not explicitly mentioned in PRIME-1576 but I'm assuming that that is the expectation?

- `CareSettingCode` is missing in the excel.
- The site needs to have organization and signing authority data together in the db to be able to show up in /admin UI.

Reply:
Org ID should be not imported as per Sarah. I don't think we are could import everyting exactly as the workflow in PRIME but only to import as much as we could.